starter-code/TicTacToe/app/src/main/java/ly/generalassemb/drewmahrt/tictactoe/GameActivity.java

```diff
@@ -1,13 +1,307 @@
 package ly.generalassemb.drewmahrt.tictactoe;
 
+import android.content.Intent;
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.GridLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+
+public class GameActivity extends AppCompatActivity{
+
+    private TextView mWhosTurn;
+
+    private TextView mBox1, mBox2, mBox3, mBox4, mBox5, mBox6, mBox7, mBox8, mBox9;
+
+    private GridLayout mGameBoard;
+
+    private Button mBackButton;
+
+    private String mP1Name;
+
+    private String mP2Name;
+
+    private boolean mP1Turn;
+
+    private ArrayList<Integer> mTestingTurnsChange;
+
+    private ArrayList<Integer> mP1Picks;
+
+    private ArrayList<Integer> mP2Picks;
+
+    private final int[][] mCombosToWin = {{1, 2, 3}, {4, 5, 6}, {7, 8, 9}, {1, 4, 7}, {2, 5, 8}, {3, 6, 9}, {1, 5, 9}, {3, 5, 7}};
 
-public class GameActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_game);
+
+        Intent recieveNameIntent = getIntent();
+        mP1Name = recieveNameIntent.getStringExtra("P1Name");
+        mP2Name = recieveNameIntent.getStringExtra("P2Name");
+
+        mWhosTurn = (TextView) findViewById(R.id.game_message_text);
+        mWhosTurn.setText(mP1Name + "'s turn");
+
+        mP2Picks = new ArrayList<>();
+        mP1Picks = new ArrayList<>();
+
+        mGameBoard = (GridLayout) findViewById(R.id.game_board);
+        mBox1 = (TextView)findViewById(R.id.textView);
+        mBox2 = (TextView)findViewById(R.id.textView2);
+        mBox3 = (TextView)findViewById(R.id.textView3);
+        mBox4 = (TextView)findViewById(R.id.textView4);
+        mBox5 = (TextView)findViewById(R.id.textView5);
+        mBox6 = (TextView)findViewById(R.id.textView6);
+        mBox7 = (TextView)findViewById(R.id.textView7);
+        mBox8 = (TextView)findViewById(R.id.textView8);
+        mBox9 = (TextView)findViewById(R.id.textView9);
+
+        mBackButton = (Button)findViewById(R.id.backButton);
+
+
+
+        mTestingTurnsChange = new ArrayList<>();
+
+        mBox1.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (mBox1.getText().toString().isEmpty()) {
```

Comment:
I suggest using one `OnClickListener` for all the TextViews so that you don't have to repeat your logic more than once